tests/token-vesting_test.ts

```diff
@@ -1,180 +1,228 @@
-    name: "[redeem] unlocked amount is transferred to the transaction sender address",
-    async fn(chain: Chain, accounts: Map<string, Account>) {
-        const deployer = accounts.get("deployer")!;
-        const txSender = accounts.get("wallet_2")!;
-        const assignees = [{address: txSender.address, amount: 250}];
-        let assigneesList: any[] = [];
-        assignees.forEach((el) => {
-            assigneesList.push(
-                types.tuple({
-                    'address': types.principal(el.address),
-                    'amount': types.uint(el.amount)
-                })
-            )
-        });
-
-        const block = chain.mineBlock(
-            [
-                Tx.contractCall(
-                    contract, "deposit",
-                    [
-                        types.principal(`${deployer.address}.${vestingOptions.token}`),
-                        types.uint(vestingOptions.amount),
-                        types.uint(vestingOptions.lockingPeriod),
-                        types.list(assigneesList)
-                    ],
-                    deployer.address
-                ),
-                Tx.contractCall(
-                    contract, "redeem",
-                    [types.principal(`${deployer.address}.${vestingOptions.token}`)],
-                    txSender.address
-                )
-            ]
-        );
-
-        const [deposit, redeem] = block.receipts;
-        deposit.result.expectOk();
-        redeem.result.expectOk();
-
-        // Check balance in the assignee address.
-        const xyzToken = new XyzToken(chain, deployer);
-        const result = xyzToken.balanceOf(txSender.address).result;
-        result.expectOk().expectUint(assignees[0].amount);
-    }
+  name: "[redeem] unlocked amount is transferred to the transaction sender address",
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get("deployer")!;
+    const txSender = accounts.get("wallet_2")!;
+    const assignees = [{address: txSender.address, amount: 250}];
+    let assigneesList: any[] = [];
+    assignees.forEach((el) => {
+      assigneesList.push(
+        types.tuple({
+          'address': types.principal(el.address),
+          'amount': types.uint(el.amount)
+        })
+        )
+    });
+
+    let block = chain.mineBlock(
+      [
+        Tx.contractCall(
+          contract, "deposit",
+          [
+          types.principal(`${deployer.address}.${vestingOptions.token}`),
+          types.uint(vestingOptions.amount),
+          types.uint(vestingOptions.lockingPeriod),
+          types.list(assigneesList)
+          ],
+          deployer.address
+        ),
+        Tx.contractCall(
+          contract, "redeem",
+          [types.principal(`${deployer.address}.${vestingOptions.token}`)],
+          txSender.address
+        )
+      ]
+    );
+
+    const [deposit, redeem] = block.receipts;
+    deposit.result.expectErr();
+    redeem.result.expectErr();
```

Comment:
I think that a token that is not whitelisted will never be subject of a `redeem`. Seems better check the whitelist guard in a separate test linked to the `deposit` feature.